src/VkLayer_GF_amber_scoop/src/draw_call_tracker.cc

```diff
@@ -48,6 +65,37 @@ void DrawCallTracker::HandleDrawCall(uint32_t first_index, uint32_t index_count,
   }
   global_data_->current_draw_call++;
 
+  auto* device_data =
+      global_data_->device_map.Get(DeviceKey(draw_call_state_.queue))->get();
+
+  VkPipelineShaderStageCreateInfo* vertex_shader = nullptr;
+  VkPipelineShaderStageCreateInfo* fragment_shader = nullptr;
+  const GraphicsPipelineData* graphics_pipeline_data =
+      device_data->graphics_pipelines.Get(draw_call_state_.graphics_pipeline)
+          ->get();
+
+  for (uint32_t stageIndex = 0;
+       stageIndex < graphics_pipeline_data->GetCreateInfo().stageCount;
+       stageIndex++) {
+    const auto& stage_create_info =
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+        graphics_pipeline_data->GetCreateInfo().pStages[stageIndex];
+    if (stage_create_info.stage == VK_SHADER_STAGE_VERTEX_BIT) {
+      vertex_shader =
+          const_cast<VkPipelineShaderStageCreateInfo*>(&stage_create_info);
+    } else if (stage_create_info.stage == VK_SHADER_STAGE_FRAGMENT_BIT) {
+      fragment_shader =
+          const_cast<VkPipelineShaderStageCreateInfo*>(&stage_create_info);
+    } else {
+      LOG("Shader stage not handled.");
+      RUNTIME_ASSERT(false);
+    }
+  }
+  // Both vertex and fragment shaders are required.
+  if (vertex_shader == nullptr || fragment_shader == nullptr) {
+    throw std::runtime_error("Missing vertex or fragment shader.");
```

Comment:
We probably also don't want to ever use exceptions. I will replace with an assert.